pkg/reconciler/instances/istio/istio_test.go

```diff
@@ -48,14 +51,124 @@ const istioctlMockCompleteVersion = `{
 		]
 	  }`
 
-func Test_RunUninstallAction(t *testing.T) {
+	istioctlMockLatestVersion = `{
+		"clientVersion": {
+		  "version": "1.11.2",
+		  "revision": "revision",
+		  "golang_version": "go1.16.7",
+		  "status": "Clean",
+		  "tag": "1.11.2"
+		},
+		"meshVersion": [
+		  {
+			"Component": "pilot",
+			"Info": {
+			  "version": "1.12.4",
+			  "revision": "revision",
+			  "golang_version": "",
+			  "status": "Clean",
+			  "tag": "1.12.4"
+			}
+		  }
+		],
+		"dataPlaneVersion": [
+		  {
+			"ID": "id",
+			"IstioVersion": "1.12.4"
+		  }
+		]
+	  }`
 
-	t.Run("Istio uninstall should also delete namespace", func(t *testing.T) {
+	istioctlMockTooNewVersion = `{
+		"clientVersion": {
+		  "version": "1.11.2",
+		  "revision": "revision",
+		  "golang_version": "go1.16.7",
+		  "status": "Clean",
+		  "tag": "1.11.2"
+		},
+		"meshVersion": [
+		  {
+			"Component": "pilot",
+			"Info": {
+			  "version": "1.13.4",
+			  "revision": "revision",
+			  "golang_version": "",
+			  "status": "Clean",
+			  "tag": "1.13.4"
+			}
+		  }
+		],
+		"dataPlaneVersion": [
+		  {
+			"ID": "id",
+			"IstioVersion": "1.13.4"
+		  }
+		]
+	  }`
+)
 
-		wsf, _ := workspace.NewFactory(nil, "./test_files", log.NewLogger(true))
+func Test_RunUpdateAction(t *testing.T) {
+	wsf, _ := workspace.NewFactory(nil, "./test_files", log.NewLogger(true))
+	model := reconciler.Task{
+		Component: "istio-configuration",
+		Namespace: "istio-system",
+		Version:   "1.11.2",
```

Comment:
Can you put here some fake Istio version? We don't have to be that specific when we put it into unit tests. It can be something like: `1.0.0` or `0.1.0`.

Reply:
It can be yes, but should we shouldn't we prefer having "real" test data configuration, as you might not detect bugs when you use fake data?